### PR TITLE
feat: add notifications settings

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description

We've added a new notifications settings section into the account screen. This section allows users to configure what notifications they wish to receive either in-app or via email, when to receive email notifications, and drops the previous notifications toggle.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/9248

## Motivation and Context

Give more control to users to decide what notifications they wish to receive.

## How Has This Been Tested?

- test environment: local & unit
- test case 1: add unit tests
- test case 2: test in the browser

## Screenshots (if appropriate):

host docker internal_9201_web-oidc-callback_code=tIgOFL3PbrHgrYTW3F7luscuS1x9lQvj scope=email%20openid%20profile session_state=b76325f6ae0240cdc6ac1f37c624fb1afda7be95b9d5885eeb881f1eb9000841 LQhHRvT0L20gbR1Me3rUY3QxLO0IE-chrnPJgQ8Fbvs stat

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-runtime/src/components/Account/AccountTable.vue [hunk 1]
```diff
@@ -4,9 +4,17 @@
...
         <oc-tr>
-          <oc-th v-for="field in fields" :key="field">{{ field }}</oc-th>
+          <template v-for="field in fields" :key="typeof field === 'string' ? field : field.label">
+            <oc-th v-if="typeof field === 'string'">{{ field }}</oc-th>
+            <oc-th
+              v-else
+              :align-h="field.alignH || 'left'"
+              :class="{ 'oc-invisible-sr': field.hidden }"
+              >{{ field.label }}</oc-th
+            >
```
Comment (saw-jan) [inline packages/web-runtime/src/components/Account/AccountTable.vue:16]:
```suggestion
            >
              {{ field.label }}
            </oc-th>
```

File: packages/web-runtime/src/components/Account/AccountTable.vue [hunk 2]
```diff
@@ -85,11 +100,13 @@ export default defineComponent({
     }
   }
 
-  td:nth-child(3) {
-    display: flex;
-    justify-content: end;
-    align-items: center;
-    min-height: var(--oc-size-height-table-row);
+  @media (min-width: 801px) {
```
Comment (saw-jan) [inline packages/web-runtime/src/components/Account/AccountTable.vue:103]: if we could use one the available breakpoints `$oc-breakpoint-`
```scss
$oc-breakpoint-large-default: 1200px;
$oc-breakpoint-large-max: 1599px;
$oc-breakpoint-medium-default: 960px;
$oc-breakpoint-medium-max: 1199px;
$oc-breakpoint-small-default: 640px;
$oc-breakpoint-small-max: 959px;
$oc-breakpoint-xlarge: 1600px;
$oc-breakpoint-xsmall-max: 639px;
```

File: packages/web-runtime/src/pages/account.vue [hunk 3]
```diff
@@ -525,6 +612,100 @@ export default defineComponent({
...
+        console.error(error)
+        showErrorMessage({
+          title: $gettext('Unable to save preference…'),
+          errors: [error]
+        })
+      }
+    }
+
+    const canConfigureSpecificNotifications = computed(
+      () => capabilityStore.capabilities.notifications.configurable
```
Comment (saw-jan) [inline packages/web-runtime/src/pages/account.vue:706]: this needs implementation in the server side, right?

Comment (LukasHirt) [reply packages/web-runtime/src/pages/account.vue:706]: Correct. For now the whole notifications settings lives in a PR https://github.com/owncloud/ocis/pull/10706

Comment (saw-jan) [reply packages/web-runtime/src/pages/account.vue:706]: ahh, great.

**Review by saw-jan — APPROVED** [review]
LGTM from my POV. :smiley:

**Comment by 2403905** [comment]
@LukasHirt Update the [#10706](https://github.com/owncloud/ocis/pull/10706)

**Comment by 2403905** [comment]
@LukasHirt Some option has the  "attribute": "disabled". Do we reflect that attribute for the checkboxes property?
```
        {
          "id": "094ceca9-5a00-40ba-bb1a-bbc7bccd39ee",
          "name": "event-space-deleted-options",
          "displayName": "Space deleted",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {}
                },
                "key": "mail",
                "attribute": "disabled",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
```

**Comment by LukasHirt** [comment]
> Some option has the "attribute": "disabled". Do we reflect that attribute for the checkboxes property?

Yes - https://github.com/owncloud/web/pull/12010/files#diff-f2ac05790ff955307ad01dea9de39462b2f5f017959394934eaad70208f0b1a7R197

**Pull request merged**